include/ensmallen_bits/utility/using.hpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * @file ensmallen_bits/utility/using.hpp
+ * @author Omar Shrit
+ * @author Ryan Curtin
+ * @author Conrad Sanderson
+ *
+ * This is a set of `using` statements to mitigate any possible risks or
+ * conflicts with local functions. The compiler is supposed to proritise the
+ * following functions to be looked up first. This is to be considered as a
+ * replacement to the ADL solution that we had deployed earlier.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENS_CORE_UTIL_USING_HPP
+#define ENS_CORE_UTIL_USING_HPP
+
+#include "function_traits.hpp"
+
+namespace ens {
+
+#ifdef USE_COOT
+
+/* using for bandicoot namespace*/
+using coot::abs;
+using coot::accu;
+using coot::chol;
+using coot::clamp;
+using coot::conv_to;
+using coot::cos;
+using coot::dot;
+using coot::exp;
+using coot::join_cols;
+using coot::join_rows;
+using coot::log;
+using coot::max;
+using coot::mean;
+using coot::min;
+using coot::norm;
+using coot::normalise;
+using coot::ones;
+using coot::pow;
+using coot::randn;
+using coot::randi;
+using coot::randu;
+using coot::regspace;
+using coot::repmat;
+using coot::shuffle;
+using coot::sign;
+using coot::size;
+using coot::sqrt;
+using coot::square;
+using coot::sum;
+using coot::trans;
+using coot::vectorise;
+using coot::zeros;
+
+#endif
+
+/* using for armadillo namespace */
+using arma::abs;
+using arma::accu;
+using arma::chol;
+using arma::clamp;
+
+// If Bandicoot is used, using arma::conv_to is already
+// part of including bandicoot.
+#ifndef USE_COOT
+using arma::conv_to;
+#endif
+
+using arma::cos;
+using arma::dot;
+using arma::exp;
+using arma::join_cols;
+using arma::join_rows;
+using arma::log;
+using arma::max;
+using arma::mean;
+using arma::min;
+using arma::norm;
+using arma::normalise;
+using arma::ones;
+using arma::pow;
+using arma::randn;
+using arma::randi;
+using arma::randu;
+using arma::regspace;
+using arma::repmat;
+using arma::shuffle;
+using arma::sign;
+using arma::size;
+using arma::sort;
```

Comment:
```suggestion
using arma::sort;
using arma::sort_index;
```
I think we need this one too so that ADL doesn't get used.